### ISSUE TITLE
Tidy XML output

### ISSUE THREAD

**Issue by idleberg** [nashwaan/xml-js#2]
I recently switched to this Node package for its versatility and I'm loving it. However, one thing I was missing is an option to format the XML output when using `js2xml()`. Consider the following example.

1.) Input object:

```js
var obj = {
  _comment: " Released under The MIT License ",
  snippet: {
    content: {
      _cdata: "console.log($1)"
    },
    tabTrigger: {
      _text: "log"
    },
    scope: {
      _text: "source.js"
    }
  }
};
```

2.) Output XML – `convert.json2xml(obj, {compact: true})`:

```xml
<!-- Released under The MIT License --><snippet><content><![CDATA[console.log($1)]]></content><tabTrigger>log</tabTrigger><scope>source.js</scope></snippet>
```

3.) Output XML – `convert.json2xml(obj, {compact: false, spaces: 4})`:

```xml
<!-- The MIT License -->
    <snippet>
        <content/>
        <tabTrigger/>
        <scope/></snippet>
```

The output in 2.) is technically valid, but I would prefer an option that produces tidy/beautified XML. Unless I did something wrong, adding the `spaces: 4` options produces an undesired result, why is both `_cdata` and `_text` missing?

Expected result (also note the indentation):

```xml
<!-- The MIT License -->
<snippet>
    <content><![CDATA[console.log($1)]]></content>
    <tabTrigger>log</tabTrigger>
    <scope>source.js</scope>
</snippet>
```

Not sure whether this is a bug or a misunderstanding of mine.

PS: I also tried converting the object to JSON with pretty-print and then use `json2xml()` instead – the results are the same!

**Comment by nashwaan** [comment]
Yes this is a bug.
I fixed this in v0.9.4.
Let me know if it is not as per your expectation.

**Comment by idleberg** [comment]
Thank you, works great!
